src/main/java/org/wso2/carbon/module/swiftiso20022/validation/rules/custom/MT103EntityFieldValidationRule.java

```diff
@@ -0,0 +1,120 @@
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.validation.rules.custom;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MT103Constants;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationRule;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidatorContext;
+
+import java.util.List;
+
+/**
+ * Entity field refers to fields which identifies an entity e.x: institution, individual
+ * Entity fields are accepted as an object with option and details keys.
+ * option is a string.
+ * details is an array of strings.
+ */
+public class MT103EntityFieldValidationRule implements ValidationRule {
+
+    private final List<ValidatorContext> contexts;
+
+    public MT103EntityFieldValidationRule(List<ValidatorContext> contexts) {
+        this.contexts = contexts;
+    }
+
+    @Override
+    public ValidationResult validate(JSONObject payload) {
+        for (ValidatorContext context : contexts) {
+
+            // validation happens only if the key is present
+            if (payload.has(context.getFieldName())) {
+
+                // entity is accepted as a JSON object
+                JSONObject entity = payload.getJSONObject(context.getFieldName());
+
+                // check whether the option is provided
+                if (entity.has(MT103Constants.MT103_ENTITY_OPTION)) {
+
+                    // option is a string
+                    String option = entity.getString(MT103Constants.MT103_ENTITY_OPTION);
+
+                    // option can either be an empty string or a single character
+                    if (option.length() > 1) {
+                        return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                                String.format(
+                                        MT103Constants.ERROR_INVALID_ENTITY_OPTION, context.getFieldDisplayName()));
+                    }
+                } else {
+                    return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                            String.format(MT103Constants.ERROR_EMPTY_ENTITY_OPTION, context.getFieldDisplayName()));
+                }
```

Comment:
Shall we move these logic to separate methods?

Reply:
Moved in commit 4e687a247135e6bc9d43274b4f9f97756cf64480